src/pages/index.js

```diff
@@ -1,94 +1,217 @@
+    "Content-Type": "application/json",
+  },
+});
 
-const imagePreview = new PopupWithImage("#popup-modal");
-imagePreview.setEventListeners();
+api
+  .loadPageContent()
+  .then(([cards, userData]) => {
+    newUserInfo.setUserAvatar(userData.avatar);
+    newUserInfo.setUserInfo({
+      name: userData["name"],
+      about: userData["about"],
+      id: userData["_id"],
+    });
+    cardSection = new Section(
+      {
+        items: cards,
+        renderer: renderCard,
+      },
+      ".cards__list"
+    );
+    cardSection.renderItems();
+  })
+  .catch((err) => {
+    console.error(err);
+  });
 
-const renderCard = (cardData) => {
-  const createCard = new Card(cardData, "#card-template", handleImageClick);
-  cardSection.addItem(createCard.getView());
-};
+let cardSection;
 
-const cardSection = new Section(
-  { items: initialCards, renderer: renderCard },
-  ".cards__list"
+const newUserInfo = new UserInfo(
+  ".profile__title",
+  ".profile__description",
+  ".profile__image"
 );
-cardSection.renderItems();
 
-const profileEditFormValidator = new FormValidator(config, profileEditForm);
-profileEditFormValidator.enableValidation();
-
-const addCardFormValidator = new FormValidator(config, addCardForm);
-addCardFormValidator.enableValidation();
-
-function handleImageClick(title, link) {
-  imagePreview.open(title, link);
-}
+/* ------------------------------ ADD NEW CARD ------------------------------ */
+const addCardPopup = new PopupWithForm("#add-card-modal", handleAddCardSubmit);
+addCardPopup.setEventListeners();
 
 function handleAddCardSubmit(data) {
-  renderCard(data);
-  addCardPopup.close();
+  addCardPopup.setLoading(true);
+  api
+    .addCard(data)
+    .then((res) => {
+      renderCard(res);
+      addCardPopup.close();
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => addCardPopup.setLoading(false, "Create"));
 }
 
+addCardOpenButton.addEventListener("click", () => {
+  formValidators["edit-profile-form"].resetValidation();
+  addCardPopup.open();
+});
+
+/* ------------------------------ EDIT PROFILE ------------------------------ */
+const editProfilePopup = new PopupWithForm(
+  "#profile-edit-modal",
+  handleProfileEditSubmit
+);
+editProfilePopup.setEventListeners();
+
 function handleProfileEditSubmit(data) {
-  newUserInfo.setUserInfo(data);
-  editProfilePopup.close();
+  editProfilePopup.setLoading(true);
+  api
+    .updateUserInfo(data.name, data.about)
+    .then((res) => {
+      newUserInfo.setUserInfo(res);
+      editProfilePopup.close();
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => editProfilePopup.setLoading(false, "Save"));
 }
 
-// Event Listeners //
-
 profileEditOpenButton.addEventListener("click", () => {
   const data = newUserInfo.getUserInfo();
   profileTitleInput.value = data.name;
-  profileDescriptionInput.value = data.description;
+  profileDescriptionInput.value = data.about;
```

Comment:
*  See here https://disk.yandex.ru/i/ujiAJcsE5ccANA when I open the profile modal the job input contains `undefined` instead of the job info